src/LowMachEquationSystem.C

```diff
@@ -425,19 +421,29 @@ LowMachEquationSystem::register_open_bc(
   bcDataAlg_.push_back(auxAlgUbc);
 
   // extract the value for user specified pressure and save off the AuxFunction
-  Pressure pSpec = userData.p_;
-  std::vector<double> userSpecPbc(1);
-  userSpecPbc[0] = pSpec.pressure_;
-
-  // new it
-  ConstantAuxFunction *theAuxFuncPbc = new ConstantAuxFunction(0, 1, userSpecPbc);
-
-  // bc data alg
-  AuxFunctionAlgorithm *auxAlgPbc
-    = new AuxFunctionAlgorithm(realm_, part,
-                               pressureBC, theAuxFuncPbc,
-                               stk::topology::NODE_RANK);
-  bcDataAlg_.push_back(auxAlgPbc);
+  if ( !realm_.solutionOptions_->activateOpenMdotCorrection_ ) {
+    ScalarFieldType *pressureBC
+      = &(metaData.declare_field<ScalarFieldType>(stk::topology::NODE_RANK, "pressure_bc"));
+    stk::mesh::put_field(*pressureBC, *part );
+    
+    Pressure pSpec = userData.p_;
+    std::vector<double> userSpecPbc(1);
+    userSpecPbc[0] = pSpec.pressure_;
+    
+    // new it
+    ConstantAuxFunction *theAuxFuncPbc = new ConstantAuxFunction(0, 1, userSpecPbc);
+    
+    // bc data alg
+    AuxFunctionAlgorithm *auxAlgPbc
+      = new AuxFunctionAlgorithm(realm_, part,
+                                 pressureBC, theAuxFuncPbc,
+                                 stk::topology::NODE_RANK);
+    bcDataAlg_.push_back(auxAlgPbc);
+  }
+  else {
+    if ( userData.pSpec_ ) 
+      throw std::runtime_error("LowMachEqs::regsiter_open_bc error: Pressure specified at an open bc while global correction algorithm has been activated");    
```

Comment:
spelling: register

Can this be a warning instead indicating that the `activate_open_mdot_correction` overrides the specified pressure at open BC?

Reply:
Absolutely. My experience is that users need hard stops to prevent bad input file propagation. That being said, I can remove the throw in favor of a warning.